### PR TITLE
[eclipse/xtext#1309] Provide Jenkinsfile for Eclipse CBI

### PR DESCRIPTION
**Pull request by kthoms**
- externalized build steps into shell scripts for convenient local usage
- added Slack notifications
- configure Maven default goal
- Make RichStringEvaluationTest test independent from system encoding
- Ignore failing tests from AbstractReusableActiveAnnotationTests
- Fix encoding issues

Signed-off-by: Karsten Thoms <karsten.thoms@itemis.de>

### CODE REVIEW

File: maven-pom.xml [hunk 1]
```diff
@@ -20,8 +20,9 @@
 	</modules>
 
 	<build>
+		<defaultGoal>deploy</defaultGoal>
```
Comment (szarnekow) [inline maven-pom.xml]: other jobs used install?

File: tycho-pom.xml [hunk 2]
```diff
@@ -31,5 +31,9 @@
 		<module>org.eclipse.xtend.doc</module>
 		<module>org.eclipse.xtend.examples</module>
 	</modules>
+	
+	<build>
+		<defaultGoal>verify</defaultGoal>
```
Comment (szarnekow) [inline tycho-pom.xml]: deploy vs install vs verify?

File: org.eclipse.xtend.core.tests/src/org/eclipse/xtend/core/tests/macro/AbstractReusableActiveAnnotationTests.xtend [hunk 3]
```diff
@@ -2922,6 +2922,7 @@ abstract class AbstractReusableActiveAnnotationTests {
 		]
 	}
 	
+	@Ignore("FIXME: Issue 1311 - Make system encoding independent")
```
Comment (cdietrich) [inline org.eclipse.xtend.core.tests/src/org/eclipse/xtend/core/tests/macro/AbstractReusableActiveAnnotationTests.xtend]: ?

File: tycho-pom.xml (hunk 2 above)
Comment (kthoms) [reply tycho-pom.xml]: for the maven plugins deploy is unfortunately necessary for now. they deploy to a local file repo, which is not nice. but different story.

File: maven-pom.xml (hunk 1 above)
Comment (kthoms) [reply maven-pom.xml]: for the maven plugins deploy is unfortunately necessary for now. they deploy to a local file repo, which is not nice. but different story.

File: tycho-pom.xml (hunk 2 above)
Comment (kthoms) [reply tycho-pom.xml]: OK, here install would be better.

**Pull request merged**